### PR TITLE
Adding an example iris training pipeline

### PR DESCRIPTION
**Pull request by DharmitD**
## The issue resolved by this Pull Request:
Resolves #<issue_number>
## Description of your changes:

Adding an example iris training pipeline, that we could point to for performance testing, or any other purposes.
## Checklist
- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: docs/example_pipelines/iris/iris-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+          \ or min_max_scaler must be True.')\n\n    import pandas as pd\n    from\
+          \ sklearn.preprocessing import MinMaxScaler\n    from sklearn.preprocessing\
+          \ import StandardScaler\n\n    with open(input_iris_dataset.path) as f:\n\
+          \        df = pd.read_csv(f)\n    labels = df.pop('Labels')\n\n    if standard_scaler:\n\
+          \        scaler = StandardScaler()\n    if min_max_scaler:\n        scaler\
+          \ = MinMaxScaler()\n\n    df = pd.DataFrame(scaler.fit_transform(df))\n\
+          \    df['Labels'] = labels\n    normalized_iris_dataset.metadata['state']\
+          \ = \"Normalized\"\n    with open(normalized_iris_dataset.path, 'w') as\
+          \ f:\n        df.to_csv(f)\n\n"
+        image: quay.io/opendatahub/ds-pipelines-iris-base:v1.0
```
Comment (amadhusu) [inline docs/example_pipelines/iris/iris-pipeline.yaml:125]: Testing this pipeline, I see an issue with accessing this image. Is it private?
 
Screenshot from 2024-02-12 18-13-45

File: docs/example_pipelines/iris/iris-pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+          _KFP_RUNTIME=true python3 -m kfp.dsl.executor_main                         --component_module_path                         "$program_path/ephemeral_component.py"                         "$@"
+
+          '
+        - "\nimport kfp\nfrom kfp import dsl\nfrom kfp.dsl import *\nfrom typing import\
+          \ *\n\ndef create_dataset(iris_dataset: Output[Dataset]):\n    import pandas\
+          \ as pd\n\n    csv_url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'\n\
+          \    col_names = [\n        'Sepal_Length', 'Sepal_Width', 'Petal_Length',\
+          \ 'Petal_Width', 'Labels'\n    ]\n    df = pd.read_csv(csv_url, names=col_names)\n\
+          \n    with open(iris_dataset.path, 'w') as f:\n        df.to_csv(f)\n\n"
+        image: quay.io/opendatahub/ds-pipelines-iris-base:v1.0
```
Comment (HumairAK) [inline docs/example_pipelines/iris/iris-pipeline.yaml:85]: can we get this dockerfile added to this path here as well

Comment (DharmitD) [reply docs/example_pipelines/iris/iris-pipeline.yaml:85]: Done, added a dockerfile

File: docs/example_pipelines/iris/iris-pipeline.yaml (hunk 1 above)
Comment (DharmitD) [reply docs/example_pipelines/iris/iris-pipeline.yaml:125]: This image isn't built yet. @HumairAK could you please copy over your image and build this one? Thanks!

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-569`

**Comment by DharmitD** [comment]
Closing this PR, duplicated in https://github.com/opendatahub-io/data-science-pipelines-operator/pull/573

**Pull request closed without merging**